### ISSUE TITLE
error "Failed to release change log lock" after Liquibase update error in SQL-Server

### ISSUE THREAD

**Issue by doakd** [liquibase/liquibase#6365]
### Search first

- [X] I searched and no similar issues were found

### Description

Looks like this issue was introduced in either 4.29.2, which we upgraded to in order to resolve the other lock removal issue.

For SQL-Server, Liquibase is unable to remove lock due to error.

command 'JAVA_OPTS="-Xmx1024m" /opt/liquibase/liquibase --changeLogFile=db-changelog-master.xml --sql-log-level=INFO --showBanner=false --log-level=INFO --defaultsFile=/tmp/tmpw5e5lcnv --classpath=/tmp/tmpfamky_wl update' returned with error (code 1): Starting Liquibase at 12:51:01 using Java 17.0.12 (version 4.29.2 #3683 built at 2024-08-29 16:45+0000)
[2024-09-26 12:51:01] INFO [liquibase.ui] Starting Liquibase at 12:51:01 using Java 17.0.12 (version 4.29.2 #3683 built at 2024-08-29 16:45+0000)
Liquibase Version: 4.29.2
[2024-09-26 12:51:01] INFO [liquibase.ui] Liquibase Version: 4.29.2
Liquibase Open Source 4.29.2 by Liquibase
[2024-09-26 12:51:01] INFO [liquibase.ui] Liquibase Open Source 4.29.2 by Liquibase
[2024-09-26 12:51:01] INFO [liquibase.integration] Starting command execution.
[2024-09-26 12:51:02] INFO [liquibase.database] Set default schema name to dbo
[2024-09-26 12:51:02] INFO [liquibase.changelog] Reading from DATABASECHANGELOG
[2024-09-26 12:51:03] INFO [liquibase.lockservice] Successfully acquired change log lock
[2024-09-26 12:51:03] INFO [liquibase.command] Using deploymentId: 7355063170
[2024-09-26 12:51:03] INFO [liquibase.changelog] Reading from DATABASECHANGELOG
.
.
<DBMS ERROR occurs>
.
.
[2024-09-26 12:51:03] INFO [liquibase.command] Update command encountered an exception.
[2024-09-26 12:51:03] WARNING [liquibase.lockservice] Failed to release change log lock
[2024-09-26 12:51:03] SEVERE [liquibase.command] Could not release lock
liquibase.exception.LockException: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Error executing SQL UPDATE DATABASECHANGELOGLOCK SET LOCKED = 0, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1: Invalid object name 'DATABASECHANGELOGLOCK'.
	at liquibase.lockservice.StandardLockService.releaseLock(StandardLockService.java:420)
	at liquibase.command.core.AbstractUpdateCommandStep.run(AbstractUpdateCommandStep.java:142)
	at liquibase.command.core.UpdateCommandStep.run(UpdateCommandStep.java:105)
	at com.datical.liquibase.ext.command.ProUpdateCommandStep.run(ProUpdateCommandStep.java:45)
	at liquibase.command.CommandScope.execute(CommandScope.java:220)
	at liquibase.integration.commandline.CommandRunner.call(CommandRunner.java:51)
	at liquibase.integration.commandline.CommandRunner.call(CommandRunner.java:21)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2045)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2465)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2457)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2419)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
	at picocli.CommandLine.execute(CommandLine.java:2174)
	at liquibase.integration.commandline.LiquibaseCommandLine.lambda$execute$2(LiquibaseCommandLine.java:404)
	at liquibase.Scope.child(Scope.java:200)
	at liquibase.Scope.child(Scope.java:176)
	at liquibase.integration.commandline.LiquibaseCommandLine.lambda$execute$3(LiquibaseCommandLine.java:379)
	at liquibase.Scope.child(Scope.java:200)
	at liquibase.Scope.child(Scope.java:176)
	at liquibase.integration.commandline.LiquibaseCommandLine.execute(LiquibaseCommandLine.java:376)
	at liquibase.integration.commandline.LiquibaseCommandLine.main(LiquibaseCommandLine.java:97)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at liquibase.integration.commandline.LiquibaseLauncher.main(LiquibaseLauncher.java:116)
Caused by: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Error executing SQL UPDATE DATABASECHANGELOGLOCK SET LOCKED = 0, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1: Invalid object name 'DATABASECHANGELOGLOCK'.
	at liquibase.executor.jvm.ChangelogJdbcMdcListener.query(ChangelogJdbcMdcListener.java:62)
	at liquibase.lockservice.StandardLockService.releaseLock(StandardLockService.java:374)

### Steps To Reproduce

Run a sql-server Liquibase CLI update command that gets an error.  

### Expected/Desired Behavior

Fails and removes change log lock

### Liquibase Version

4.29.2

### Database Vendor & Version

MICROSOFT SQL SERVER 2022 SPRTM ENTERPRISE (16.0.4135.4)	

### Liquibase Integration

CLI

### Liquibase Extensions

oracle, teradata, databricks

### OS and/or Infrastructure Type/Provider

Debian

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by tati-qalified** [comment]
Hi @doakd, thank you for reporting this issue.
Was your database empty before running `update`? Or did the DBCLL table already exist?
Do you encounter the same problem when the `update`  command should not return an error?

**Comment by tati-qalified** [comment]
What type of error is causing the update to fail in this case? Are you having this problem with different causes?
I tested it with non-existing tables and invalid column names but couldn't replicate it.

**Comment by doakd** [comment]
Any dbms failure will cause the lock to be left, though that error message is not always present in the output.

* Insert with PK violation
* create where object already exists
* etc.

1. run an "update" command that fails
2. immediately run another "update" command, it will be blocked with lock in place.

**Comment by doakd** [comment]
Please upvote.  This is occurring for every failed SQL in SQL-Server.  Lock is left in place.    My customer is forced to run "release-locks" after every failed "update".

Lock release error is not always in the output log.

**Comment by molivasdat** [comment]
@doakd 
Are you using the default name for the databasechangeloglock table?
Is your database case sensitive or insensitive on SQL Server?
Is the dbcll table upper or lowercase?

**Comment by doakd** [comment]
Yes, default table name.

Databases are case insensitive.

dbcll table is uppercase.

**Comment by lbadam** [comment]
@doakd would you be able to join us for a call so we could take a look at the issue together? If so, feel free to find a time [on my calendar](https://calendly.com/amurray-liquibase/debug-session) or let me know a time that works for you.

**Comment by doakd** [comment]
ok, I think I got to root cause of the issue.

I found some SQL-Server execution that failed, but did not have this dblll lock release issue, so I did a deeper comparison.

I found that the customer having this issue is putting "USE" statements at the top and bottom of their changelogs, so if there is a SQL snytax issue, the final "USE" statement is never executed, resulting in Liquibase trying to update the dbcll table in the wrong database, where it does not exist.

I am making a recommendation to them to not use "USE" anymore, and instead prefix the tables with the required database name.

Sorry for the firedrill.
